app/src/main/java/org/adaway/helper/ImportExportHostsHelper.java

```diff
@@ -0,0 +1,294 @@
+        import java.util.Set;
+
+/**
+ * This class is a helper class to import/export user lists to a backup file on sdcard.
+ *
+ * @author Bruce BUJON (bruce.bujon(at)gmail(dot)com)
+ */
+public class ImportExportHostsHelper {
+    /**
+     * The request code to identify the selection of a file in {@link androidx.fragment.app.Fragment#onActivityResult(int, int, Intent)}.
+     */
+    public final static int REQUEST_CODE_IMPORT = 45;
+
+    /**
+     * Import the given user lists backup file.
+     *
+     * @param context      The application context.
+     * @param userListsUri The URI of the user lists backup file.
+     */
+    public static void importLists(Context context, Uri userListsUri) {
+        // Import user lists
+        new ImportListsTask(context).execute(userListsUri);
+    }
+
+    /**
+     * Exports all lists to adaway-export file on sdcard.
+     *
+     * @param context The application context.
+     */
+    public static void exportLists(final Context context) {
+        // Export user lists
+        new ExportListsTask(context).execute();
+    }
+
+    /**
+     * This class is an {@link AsyncTask} to import user lists.
+     *
+     * @author Bruce BUJON (bruce.bujon(at)gmail(dot)com)
+     */
+    private static class ImportListsTask extends AsyncTask<Uri, Void, Void> {
+        /**
+         * A weak reference to application context.
+         */
+        private final WeakReference<Context> mWeakContext;
+        /**
+         * The progress dialog.
+         */
+        private ProgressDialog mProgressDialog;
+
+        /**
+         * Constructor.
+         *
+         * @param context The application context.
+         */
+        private ImportListsTask(Context context) {
+            // Store context into weak reference to prevent memory leak
+            this.mWeakContext = new WeakReference<>(context);
+        }
+
+        @Override
+        protected Void doInBackground(Uri... results) {
+            // Check parameters
+            if (results.length < 1) {
+                return null;
+            }
+            // Get URI to export lists
+            Uri result = results[0];
+            // Get context from weak reference
+            Context context = this.mWeakContext.get();
+            if (context == null) {
+                return null;
+            }
+            // Get database
+            AppDatabase database = AppDatabase.getInstance(context);
+            // Get input stream from user selected URI
+            try (InputStream inputStream = context.getContentResolver().openInputStream(result)) {
+                if (inputStream != null) {
+                    // Create reader from input stream
+                    try (BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream))) {
+                        Set<String> newHosts = new HashSet<>();
+                        String line = reader.readLine();
+                        while (line != null) {
```

Comment:
You can have a single line like `while ((line = reader.readLine()) != null) {`

Reply:
I wasnt sure if that would catch the first line? I guess it should do. Will change and push